Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -452,12 +471,19 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
     }
 
     private func numberOfColumnsForCurrentSize() -> Int {
-        if self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClass.Compact {
-            return TabTrayControllerUX.NumberOfColumnsRegular
-        } else if self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClass.Compact {
-            return TabTrayControllerUX.NumberOfColumnsCompact
+        let compactLayout = profile.prefs.boolForKey("CompactTabLayout") ?? true
+        let idiom = UIDevice.currentDevice().userInterfaceIdiom
+        let orientation = UIDevice.currentDevice().orientation
+
+        if idiom == .Phone {
```

Comment:
Since we're currently using the callbacks for rotation changes in the TabTrayController, let's keep checking the device instead of traitCollection. I think some of the changes I have in my refactor branch will affect this but I can deal with those changes.